ProcessOptimizer/model_systems/noise_models.py

```diff
@@ -1,33 +1,47 @@
 from abc import ABC, abstractmethod
 from typing import Callable, List, Union, Optional
 
-from scipy.stats import norm, uniform
+import numpy as np
 
 class NoiseModel(ABC):
     """
     Abstract class that is the basis for noise models.
     """
     def __init__(
-            self,
-            noise_size: Optional[float]):
+        self,
+        noise_size: Optional[float],
+        seed: Union[int, None] = 42,
+    ):
         """        
-        Parameters:
-        * `noise_size` [Option[float]]: The size of the noise. If ´None´, it signifies that
-            the class is compound, and should not have its own noise signal, but refer to
-            its compounding NoiseModels instead. An example is SumNoise, which sums the
-            noise of a list of noise models, but doesn't add any noise by itself."""
+        Parameters
+        ----------
+        * `noise_size` [Optional[float]]: 
+            The size (magnitude) of the noise. If ´None´, it signifies that
+            the class is compound, and should not have its own noise signal, 
+            but refer to its compounding NoiseModels instead. An example is 
+            SumNoise, which sums the noise of a list of noise models, but does
+            not add any noise by itself.
+        
+        * `seed` [Union[int, None], default=42]:
+            Seed to pass forward to the numpy random number generator that is
+            used when providing samples with noise.
+        """
         # We could just set noise_dist and let that have a size, but being able to
         # directly set the size is more intuitive, and that would be complicated if it
         # was just one variable.
         # Note that this has the potential for problems if _noise_distribution does not
         # have "size" 1, but as long as it is only set by set_noise_type(), it should be
         # safe.
         self.noise_size = noise_size
-        self._noise_distribution: Callable[[],float]
+        if seed != None:
```

Comment:
The default seed for np.random.default_rng is `None`, so you don't need this if...else branching, you can just use
```
self._rng = np.random.default_rng(seed)
```

Reply:
Implemented.